qiskit/providers/fake_provider/fake_generic.py

```diff
@@ -0,0 +1,546 @@
+            self._target.add_instruction(BreakLoopOp, name="break")
+            self._target.add_instruction(ContinueLoopOp, name="continue")
+
+    def _add_noisy_instruction_to_target(
+        self,
+        instruction: Instruction,
+        noise_params: tuple[float, ...] | None,
+        calibration_inst_map: InstructionScheduleMap | None,
+    ) -> None:
+        """Add instruction properties to target for specified instruction.
+
+        Args:
+            instruction: Instance of instruction to be added to the target
+            noise_params: Error and duration noise values/ranges to
+                include in instruction properties.
+            calibration_inst_map: Instruction schedule map with calibration defaults
+        """
+        qarg_set = self._coupling_map if instruction.num_qubits > 1 else range(self.num_qubits)
+        props = {}
+        for qarg in qarg_set:
+            try:
+                qargs = tuple(qarg)
+            except TypeError:
+                qargs = (qarg,)
+            duration, error = (
+                noise_params
+                if len(noise_params) == 2
+                else (self._rng.uniform(*noise_params[:2]), self._rng.uniform(*noise_params[2:]))
+            )
+            if (
+                calibration_inst_map is not None
+                and instruction.name not in ["reset", "delay"]
+                and qarg in calibration_inst_map.qubits_with_instruction(instruction.name)
+            ):
+                # Do NOT call .get method. This parses Qobj immediately.
+                # This operation is computationally expensive and should be bypassed.
+                calibration_entry = calibration_inst_map._get_calibration_entry(
+                    instruction.name, qargs
+                )
+            else:
+                calibration_entry = None
+            props.update({qargs: InstructionProperties(duration, error, calibration_entry)})
+        self._target.add_instruction(instruction, props)
+
+        # The "measure" instruction calibrations need to be added qubit by qubit, once the
+        # instruction has been added to the target.
+        if calibration_inst_map is not None and instruction.name == "measure":
+            for qarg in calibration_inst_map.qubits_with_instruction(instruction.name):
+                try:
+                    qargs = tuple(qarg)
+                except TypeError:
+                    qargs = (qarg,)
+                # Do NOT call .get method. This parses Qobj immediately.
+                # This operation is computationally expensive and should be bypassed.
+                calibration_entry = calibration_inst_map._get_calibration_entry(
+                    instruction.name, qargs
+                )
+                for qubit in qargs:
+                    self._target[instruction.name][(qubit,)].calibration = calibration_entry
+
+    def run(self, run_input, **options):
+        """Run on the backend using a simulator.
```

Comment:
Minor, I know this is what is done in `FakeBackendV2`, but IMO it's usually better to include documentation like this in the class docstring to explain how this class satisfies the behavior of `BackendV2` (e.g. that it uses specific simulators, etc.).

The `run` method should always do exactly what the docs say on the base class from an API perspective. If it doesn't, it's a broken abstraction. As a user, having a redundant explanation of the args and return value makes me do a double-take, especially since this is already out of sync with what's documented in `BackendV2`.

I'm not requesting any change here, but just offering a viewpoint counter to that of whoever wrote `FakeBackendV2` 🙂.